internal/authentication/tokengetter.go

```diff
@@ -86,6 +96,10 @@ func (t *TokenGetter) getToken(ctx context.Context, key types.NamespacedName) (*
 			Spec: authenticationv1.TokenRequestSpec{ExpirationSeconds: ptr.To(int64(t.expirationDuration / time.Second))},
 		}, metav1.CreateOptions{})
 	if err != nil {
+		if errors.IsNotFound(err) {
+			saErr := &ServiceAccountNotFoundError{key.Name}
+			return nil, saErr
+		}
```

Comment:
```suggestion
		if errors.IsNotFound(err) {
			return nil, &ServiceAccountNotFoundError{ServiceAccountName: key.Name}
		}
```

Define the attribute name explicitly, which the value represents. This improves the readability of the code
We can also return directly